packages/angular/src/generators/move-to-with-mf/move-to-with-mf.ts

```diff
@@ -0,0 +1,50 @@
+import type { Tree } from '@nrwl/devkit';
+import type { Schema } from './schema';
+
+import { readProjectConfiguration, formatFiles } from '@nrwl/devkit';
+import { getMfeProjects } from '../../utils/get-mfe-projects';
+import {
+  checkOutputNameMatchesProjectName,
+  checkSharedNpmPackagesMatchExpected,
+  getWebpackConfigPath,
+  isHostRemoteConfig,
+  parseASTOfWebpackConfig,
+  writeNewWebpackConfig,
+} from './lib';
+
+export default async function moveToWithMF(tree: Tree, schema: Schema) {
+  const projects = new Set(getMfeProjects(tree));
+
+  if (!projects.has(schema.project)) {
+    throw new Error(
+      `Could not find project \`${schema.project}\` in your workspace. Please check the name of the project you're wishing to convert exists.`
+    );
+  }
+
+  const project = readProjectConfiguration(tree, schema.project);
+  const pathToWebpackConfig = getWebpackConfigPath(project, schema.project);
+  const webpackAst = parseASTOfWebpackConfig(tree, pathToWebpackConfig);
+
+  if (!checkOutputNameMatchesProjectName(webpackAst, schema.project)) {
+    throw new Error(
+      `Cannot automatically migrate \`${schema.project}\` to \`withModuleFederation\` mirco frontend webpack config. 
+      \`uniqueName\` in webpack config (${pathToWebpackConfig}) does not match project name.`
+    );
+  }
+
+  if (!checkSharedNpmPackagesMatchExpected(webpackAst)) {
+    throw new Error(
+      `Cannot automatically migrate \`${schema.project}\` to \`withModuleFederation\` mirco frontend webpack config. 
+        There are npm packages being shared with a custom configuration in webpack config (${pathToWebpackConfig}).`
```

Comment:
I think we should remove `npm` here, it can also be workspace packages/libs.
```suggestion
      `Cannot automatically migrate "${schema.project}" to "withModuleFederation" micro frontend webpack config. 
        There are packages being shared with a custom configuration in webpack config (${pathToWebpackConfig}).`
```